Alignment/OfflineValidation/plugins/PrimaryVertexValidation.cc

```diff
@@ -1097,27 +1157,29 @@ void PrimaryVertexValidation::analyze(const edm::Event& iEvent, const edm::Event
 }
 
 // ------------ method called to discriminate 1D from 2D hits  ------------
-bool PrimaryVertexValidation::isHit2D(const TrackingRecHit& hit) const {
+bool PrimaryVertexValidation::isHit2D(const TrackingRecHit& hit, const PVValHelper::detectorPhase& thePhase) const {
   if (hit.dimension() < 2) {
     return false;  // some (muon...) stuff really has RecHit1D
   } else {
     const DetId detId(hit.geographicalId());
     if (detId.det() == DetId::Tracker) {
       if (detId.subdetId() == PixelSubdetector::PixelBarrel || detId.subdetId() == PixelSubdetector::PixelEndcap) {
-        return true;  // pixel is always 2D
-      } else {        // should be SiStrip now
+        return true;                                 // pixel is always 2D
+      } else if (thePhase != PVValHelper::phase2) {  // should be SiStrip now
         if (dynamic_cast<const SiStripRecHit2D*>(&hit))
           return false;  // normal hit
         else if (dynamic_cast<const SiStripMatchedRecHit2D*>(&hit))
           return true;  // matched is 2D
         else if (dynamic_cast<const ProjectedSiStripRecHit2D*>(&hit))
           return false;  // crazy hit...
         else {
-          edm::LogError("UnkownType") << "@SUB=AlignmentTrackSelector::isHit2D"
+          edm::LogError("UnkownType") << "@SUB=PrimaryVertexValidation::isHit2D"
```

Comment:
"Unknown" is misspelled. This misspelling is present in several places in CMSSW: https://github.com/cms-sw/cmssw/search?q=UnkownType I suggest changing the misspellings all at once but will sign off on the PR anyway. You can decide if you want to fix them now or later; I don't have an opinion.

Reply:
@christopheralanwest 
it's addressed at https://github.com/cms-sw/cmssw/pull/32432